spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryAutoConfiguration.java

```diff
@@ -36,19 +36,16 @@
 
 /**
  * @author Spencer Gibb
+ * @author Tim Ysewyn
  * @since 1.1.0
  */
 @Configuration
-@ConditionalOnBean(ZookeeperDiscoveryClientConfiguration.Marker.class)
```

Comment:
Why is this gone?

Reply:
Because now it's being used by both `ZookeeperDiscoveryClientConfiguration` and `ZookeeperReactiveDiscoveryClientConfiguration`. The `zookeeperDiscoveryClient` has been moved to `ZookeeperDiscoveryClientConfiguration` so it has a more general purpose.

Reply:
So the marker isn't needed anymore?

Reply:
So is `ZookeeperDiscoveryClientConfiguration` auto configuration now? The reason no beans were put there is because enable discovery client imported that class and it didn't follow auto configuration rules and ordering

Reply:
Interesting. I added those 2 configuration files to `org.springframework.cloud.client.discovery.EnableDiscoveryClient` in the `spring.factories` file and added the `@AutoConfigureBefore({ ZookeeperDiscoveryAutoConfiguration.class })` annotation. That seems to work. I kept the marker bean for backwards compatibility, other projects might use that in a `@ConditionalOnBean` too.

Reply:
I think it is by luck or the fact that they are listed as auto configuration. https://github.com/spring-cloud/spring-cloud-zookeeper/pull/235/files#diff-de1b64397dfb792529ca611c453ebdd9R13

We need to move back to a model where the ones that are listed under `EnableDiscoveryClient` only create marker beans and the auto config is conditional on the marker beans. It adds an extra class for each.